src/Gizmos/gizmoMaterialSwitcher.ts

```diff
@@ -0,0 +1,197 @@
+import { PointerDragBehavior } from "../Behaviors/Meshes/pointerDragBehavior";
+import { PointerInfo } from "../Events/pointerEvents";
+import { HemisphericLight } from "../Lights/hemisphericLight";
+import { StandardMaterial } from "../Materials/standardMaterial";
+import { Color3 } from "../Maths/math.color";
+import { AbstractMesh } from "../Meshes/abstractMesh";
+import { Scene } from "../scene";
+
+/**
+ * Tracks interactivity state of a Gizmo and switches materials of it's meshes
+ */
+export class GizmoMaterialSwitcher {
+    /** @hidden */
+    /**
+     * Tracks Scenes that have an active Gizmo drag interaction.
+     * Assumes only one Gizmo can be dragged at a time per Scene
+     */
+    public static _ScenesWithActiveDrag: Scene[] = [];
+
+    private _dragBehavior: PointerDragBehavior;
+    private _sharedGizmoLight: HemisphericLight;
+    private _scene: Scene;
+
+    private _defaultMaterial: StandardMaterial;
+    private _hoverMaterial: StandardMaterial;
+    private _dragMaterial: StandardMaterial;
+
+    private _isHover: boolean = false;
+    private _isDrag: boolean = false;
+    private _material: StandardMaterial;
+
+    private _meshes: AbstractMesh[];
+
+    /**
+     * Creates a GizmoMaterialSwitcher
+     * @param color The color of the gizmo
+     * @param dragBehavior PointerDragBehavior of the Gizmo
+     * @param sharedGizmoLight The Light from gizmoLayer._getSharedGizmoLight()
+     * @param scene Scene from gizmoLayer.utilityLayerScene
+     */
+    constructor(
+        color: Color3,
+        dragBehavior: PointerDragBehavior,
+        sharedGizmoLight: HemisphericLight,
+        scene: Scene
+    ) {
+        this._dragBehavior = dragBehavior;
+        this._sharedGizmoLight = sharedGizmoLight;
+        this._scene = scene;
+
+        // Create Materials
+        this._defaultMaterial = new StandardMaterial("_defaultMaterial", scene);
```

Comment:
is it possible to override these values? I noticed that the default material is exposed but maybe we can add an optional default material in the constructor options?

Reply:
I am also a bit concerned restricting this to just StandardMaterial. But is there a likely need for users to have an easy way to use a different material class instead of just changing properties of these StandardMaterials? Would this even work in the utility layer with that single light?

If that is a realistic usecase then I would prefer not changing the signature but expose a static MaterialConstructor with the default implementation returning these standardMaterials. But then I would likely need to refactor for those to materials to be kept together in a single data structure.

Also, is there a real need for 3 materials instead altering the color of a single material. I see that the default material has specularity color but besides color otherwise they are identical.

Reply:
due to backwards compatibility we cannot remove any public members. If you think however that it can be the same material, go right ahead and set those properties to return a single material. 

I don't think it needs to be more than a standard material, but it should be accessible for the user that's what I meant with overriding the values. As simple as possible, nothing too complex.